src/mon/OSDMonitor.cc

```diff
@@ -7891,7 +7899,7 @@ bool OSDMonitor::preprocess_pool_op(MonOpRequestRef op)
     }
     return false;
   case POOL_OP_DELETE_SNAP:
-    if (p->is_unmanaged_snaps_mode()) {
+    if (p->is_unmanaged_snaps_mode() || p->is_tier()) {
```

Comment:
it's not necessary to check if `p->is_tier()` again? if it is not allowed to create a snap for a tier pool?